src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -682,8 +685,9 @@ private int CalculateScanlineLength(int width)
     /// <param name="image"> The pixel data.</param>
     /// <param name="pngMetadata">The png metadata</param>
     /// <param name="getData">A delegate to get more data from the inner stream for <see cref="ZlibInflateStream"/>.</param>
+    /// <param name="frameControl">The frame control</param>
     /// <param name="cancellationToken">The cancellation token.</param>
-    private void ReadScanlines<TPixel>(int chunkLength, ImageFrame<TPixel> image, PngMetadata pngMetadata, Func<int> getData, CancellationToken cancellationToken)
+    private void ReadScanlines<TPixel>(int chunkLength, ImageFrame<TPixel> image, PngMetadata pngMetadata, Func<int> getData, FrameControl frameControl, CancellationToken cancellationToken)
```

Comment:
Pass `FrameControl` via `in` to avoid copying.

Reply:
Do I need to put `in` before all `FrameControl`'s? Same as here.